packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -1594,37 +1619,56 @@ namespace Ifpack2 {
       KOKKOS_INLINE_FUNCTION 
       void 
       extract(const member_type &member, 
-              const local_ordinal_type &partidx,
-	      const local_ordinal_type &v) const {
-        const size_type kps = pack_td_ptr(partidx);
-        const local_ordinal_type kfs = flat_td_ptr(partidx);
-        const local_ordinal_type ri0 = partptr(partidx);
-        const local_ordinal_type nrows = partptr(partidx+1) - ri0;
-        for (local_ordinal_type tr=0,j=0;tr<nrows;++tr) {          
-          local_ordinal_type lbeg = (tr == 0         ? 1 : 0);
-          local_ordinal_type lend = (tr == nrows - 1 ? 2 : 3);
-          for (local_ordinal_type l=lbeg;l<lend;++l,++j) { // l == 1, diagonal
-            const size_type Aj = A_rowptr(lclrow(ri0 + tr)) + A_colindsub(kfs + j);
-            const impl_scalar_type* block = &A_values(Aj*blocksize_square);
-            const size_type pi = kps + j;
-            Kokkos::parallel_for
-              (Kokkos::TeamThreadRange(member,blocksize_square), [&](const local_ordinal_type &ij) {
-                const local_ordinal_type ii = ij/blocksize;
-                const local_ordinal_type jj = ij%blocksize;
-                scalar_values(pi, ii, jj, v) = block[ii*blocksize + jj];
-              });
+              const local_ordinal_type &partidxbeg,
```

Comment:
It's better to pass by value than by reference here, since the values have smaller type.  That won't matter if inlining works, but it's good to get into the habit of passing by value where it makes sense.

Reply:
I agree.